### ISSUE TITLE
Caret event fired if moving review cursor when UIA disabled in ms word

### ISSUE THREAD

**Issue by burmancomp** [nvaccess/nvda#17227]
### Steps to reproduce:
- disable UIA for ms word in nvda advanced settings
- open ms word and write some lines (likely result is same when opening existing document)
- move review cursor up and down
### Actual behavior:

Nvda gets caret event
### Expected behavior:

As far as I understand, no caret event should be fired.
### NVDA logs, crash dumps and other attachments:
IO - inputCore.InputManager.executeGesture (22:46:04.203) - winInputHook (10652):
Input: kb(laptop):NVDA+upArrow
...
DEBUG - IAccessibleHandler.winEventToNVDAEvent (22:46:04.383) - MainThread (16360):
winEvent mapped to NVDA event: caret
...
IO - inputCore.InputManager.executeGesture (22:46:06.602) - winInputHook (10652):
Input: kb(laptop):NVDA+downArrow
...
DEBUG - IAccessibleHandler.winEventToNVDAEvent (22:46:06.732) - MainThread (16360):
winEvent mapped to NVDA event: caret

#### NVDA installed/portable/running from source:
installed
#### NVDA version:
alpha-34161,1573904b (2025.1.0.34161)
#### Windows version:
windows 10 22h2/windows 11 23h2
#### Name and version of other software in use when reproducing the issue:
word 2019
#### Does the issue still occur after restarting your computer?
yes
#### If NVDA add-ons are disabled, is your problem still occurring?
yes

**Comment by Adriani90** [comment]
cc: @michaelDCurran do you know if this was intended? I think due to the nature of objectModel it was required in order to scroll the screen when using browse mode, but not sure.

**Comment by CyrilleB79** [comment]
@Adriani90, to clarify and just to be sure that we all speak of the same thing:
Moving the review cursor and moving the browse mode cursor is not the same thing. By the way, the browse mode cursor only exists in browsers, where there is no real cursor (no caret). In Word, there is a real cursor (caret); so even in browse mode, using arrows moves the caret.

**Comment by Adriani90** [comment]
@burmancomp Are you saying with UIA enabled this is not the case?
Do you have navigator follows system caret enabled in NVDA review cursor settings?

**Comment by seanbudd** [comment]
This is an issue with MS Word. Please consider that this is hidden behind advanced preferences and is not encouraged. Closing as invalid

**Comment by Adriani90** [comment]
This needs clarification, reopening.

**Comment by XLTechie** [comment]
@seanbudd

> This is an issue with MS Word. Please consider that this is hidden behind advanced preferences and is not encouraged.

It seems strange for NV Access to consider UIA disablement for word to be "not encouraged", given how many important features of Word, simply fail to work correctly with it enabled.
Users turning UIA off should be no irregular or unexpected thing.

If it's only that it's under "advanced", then I suggest moving it to a less controversial panel.

The problems with UIA, and thus the reasons why users need to turn it off, are well documented.

I can't imagine that NV Access is reluctant to recognize this.

**Comment by CyrilleB79** [comment]
Given the real usage of Word UIA option, may we consider moving this option in a non-advanced settings panel? It's worth noting that this option may not cause any serious issue. It may degrade performances if legacy is selected instead of UIA. But we have at least another such option in non advanced settings panel: "Report formatting changes after the cursor (can cause a lag)" in document formatting settings.

Of course, ideally, NVDA should be able to request information to various API (UIA, IAccessible, object model) according to the situation without having to ask the user which interface is needed; e.g.from what I know, there is nosuch setting in other screen readers. But since it's not the case and won't probably be soon, let's find another solution.

For reference, Word UIA issues have been discussed in #16600, where @Adriani90 and others have listed the limitations of this mode.

IMO NV Access will have to discuss this internally. Cc @gerald-hartig

**Comment by Emil-18** [comment]
> Of course, ideally, NVDA should be able to request information to various API (UIA, IAccessible, object model) according to the situation without having to ask the user which interface is needed; e.g.from what I know, there is nosuch setting in other screen readers.

You can turn off UIA support for word in JAWS.
Doesn't NVDA already decide if it should use UIA or not based on the office version in use?

**Comment by CyrilleB79** [comment]
> You can turn off UIA support for word in JAWS.

Oh, didn't know. This confirms that moving the Use UIA with Word option out of the settings panel may be a good idea.

> Doesn't NVDA already decide if it should use UIA or not based on the office version in use?

Yes, by default, NVDA uses UIA in Word (and Outlook) for Microsoft Office 16.0.15000 or higher and only in Windows 11.

Though, some Windows 10 users sometimes enable UIA in MS Word anyway to have better performances in big documents, even if there is probably a limitation elsewhere (ability to read comments, access to math equations?). On the other side, some users of Windows 11 and Word > 15000 sometimes select "Only if necessary" to fall back to object model, because they encounter one of the limitations with UIA (see #16600).

**Comment by XLTechie** [comment]
> Doesn't NVDA already decide if it should use UIA or not based on the office version in use?

Yes, but that doesn't consider the various cases where UIA doesn't lead to the same level of accessibility that the older mechanisms do.

It is just using UIA because it's there, not because it works better across all Word features.
